### ISSUE TITLE
Consolidate NixCon pages

### ISSUE THREAD

**Issue by davidak** [NixOS/nixos-homepage#397]
# Original state

We had 4 official NixCons and their pages are on 4 different domains:

- conf.nixos.org
- nixcon2017.org
- nixcon2018.org
- 2019.nixcon.org

Can we get access to nixcon.org?

It would be great to have:

- https://nixcon.org/ (announcement of and link to next planned conference, links to previous)
- https://nixcon.org/2020 (with simple announcement "a conference is planned. help \<link to discourse thread\>")
- https://nixcon.org/2019
- https://nixcon.org/2018
- https://nixcon.org/2017
- https://nixcon.org/2015

See https://fosdem.org/2020/ for example.

Maintaining the domain and webspace is probably a task for the infrastructure team. The conference organizers have to upload the pages (maybe in a git repo?)

**Comment by garbas** [comment]
pinging current nixcon team members /cc @andir @zimbatm @flokli

**Comment by flokli** [comment]
Yeah, the current situation is a bit chaotic - so far, we repurposed the old website for the next years nixcon, by forking and editing it ;-)

Moving everything below specific subpaths might introduce too much churn in updating references - so I'd prefer a subdomain for each year. This would also allow to change ways how we host things (GitHub Pages vs. another service vs. hosting by ourselves).

**Comment by zimbatm** [comment]
Curently we have one sub-domain per year. The advantage with that approach is that each year is completely independent and can use a different static generate and template.

eg: http://2019.nixcon.org/

You can find all the sites at https://github.com/nixcon/

**Comment by davidak** [comment]
>so far, we repurposed the old website for the next years nixcon, by forking and editing it ;-)

that's totally fine. FOSDEM seems to do that too and they are a lot bigger and more popular.

>Moving everything below specific subpaths might introduce too much churn in updating references - so I'd prefer a subdomain for each year.

theoretically it should just be 1 sed command, right? i think we should invest time here when it improves visitors experience. i think a subpath is better symantically. NixCon 2020 has the URL https://nixcon.org/2020 (you can read it from left to right. and search engines probably count it as one site and get a better rank, not that we need more participants :D)

>This would also allow to change ways how we host things (GitHub Pages vs. another service vs. hosting by ourselves).

that is a point, but is that really a problem we need to solve? we can easily host all static sites in one webspace at one hoster or migrate them to a new one. actually any webhoster should work here

>The advantage with that approach is that each year is completely independent and can use a different static generate and template.

you can easily have that with subfolders. deploy (aka upload) new site to `/2020` instead of `/` and have just one `index.html` in `/`

you could even have a different CMS in every subfolder. Wordpress for `/2020` and Drupal for `/2019`. any webhoster with PHP should support that. they cost [1,90€ a month](https://www.hetzner.de/webhosting) or so and we with NixOS should be able to host that ourselves... we have all options with that approach, but i also prefer static sites because they are easy to maintain (imagine having to update a Joomla in `/2015` every month :D)

where are the sites currently hosted? should i create a demonstration of my suggested approach?

**Comment by zimbatm** [comment]
At the moment each site is its own repo in https://github.com/nixcon/ and deployed to GitHub Pages as a static website. The DNS is managed with CloudFlare and access can be shared. All of this is free and maintains itself.

We also have a bit of infrastructure deployed with https://github.com/nixcon/nixcon-vps for ticket sales and CfP hosted on vpsfree that we have to keep up to date.

Currently, the maintenance is `O(1)` over the number of years which is a feature I think is desirable. Once a conference is done we should upload the videos and then keep it essentially archived. If you want to push all the sites into a single GitHub Page with each their own prefix that's also fine by me.

**Comment by davidak** [comment]
>Currently, the maintenance is O(1) over the number of years which is a feature I think is desirable. Once a conference is done we should upload the videos and then keep it essentially archived.

I agree. You should be able to focus on organizing an amazing conference :)

>If you want to push all the sites into a single GitHub Page with each their own prefix that's also fine by me.

what do you mean by "prefix"? would that result in `https://nixcon.org/2020`?

I haven't used GitHub Page yet, but will look what can be done when i find some time for it. Thanks!

**Comment by garbas** [comment]
I would also go with the proposal @zimbatm suggested, one repo -> static html -> served via github pages -> subdomain (YEAR.nixcon.org).

1. Should we also archive the videos somewhere or is it ok to leave them on YouTube?
2. And as we have all video in one place, I can help move videos from berlin nixcon over? Anybody knows how to do this?
3. If you want we can also use NixOS channel for all of this (probably create a separate subchannel or just simple playlists) https://www.youtube.com/channel/UC3vIimi9q4AT8EgxYp_dWIw ... It would be nice to have one place to go for all NixOS related videos.

**Comment by davidak** [comment]
>Should we also archive the videos somewhere or is it ok to leave them on YouTube?

a backup does not hurt

>And as we have all video in one place, I can help move videos from berlin nixcon over? Anybody knows how to do this?

Over to youtube? There is actually a playlist on the account of the C3VOC which have done the streaming and recording.

https://www.youtube.com/playlist?list=PL_IxoDz1Nq2Y7mIxMZ28mVtjRbbnlVdmy

we can probably just add them to our playlist. it don't make much sense to have the same video multiple times on youtube where our would be less popular

>If you want we can also use NixOS channel for all of this

it would be perfect to have every official video on one NixOS account with proper playlists

**Comment by andir** [comment]
IMO just having `${year].nixcon.org` is fine. It allows us to change the entire technology stack over the years without having to migrate content that someone 10y ago wrote in some Haskell/Go/JS/… framework that nobody understands but still works well enough in current browsers.

We should redirect from nixcon.org/${year} to that year (as it is right now). The landing page can also be a redirect to the current year.

Regarding recordings:

We should have a backup of the files. Those should either be the raw recording or the final cut version - either is better than nothing. Important here is that we do not just upload it to yet another cloud provider where we might be able to extract it now (e.g. youtube-dl) but to some place where we can download it even in 5 or 10y. Thus I think having them archived on a non-video-streaming site is probably the way to go. During/After last years NixCon I briefly thought about the idea and we did look for some place to store them but didn't come up with a solution. For now I have a copy of all the recordings on my NAS.

Compared to the size of the nixos.org binary cache these recordings are tiny. One way to solve this would probably be to put them in a bucket on S3 just like our binary cache.. It is mostly filesystem like and it is made to be accessible. Ideally a "permanent" entity would be billed for these charges. Not because of the money but because of dedication to the project. We don't want those to be lost just because someone doesn't do computers anymore and moved to the woods.

Publishing recordings on a single platform is another topic. While generally I like to "not feed the google". In this case YouTube seems like the canonical place for those (while we keep an off-google copy). The reason YT isn't that bad is because for now every year the location (and parts of the team) around NixCon have changed. Not being tied to some localized group of people (e.g. c3voc^[1]) for our videos (and streaming) is probably a win here. Less overhead for those organizing and running the show.



[1]: I like them. We are/would be using their services for NixCon2020 if COVID-19 permits.

**Comment by zimbatm** [comment]
The logic could also work if every repo were to push to their own prefix instead of having one sub-domain per year. As long as each year is not dependent on the previous ones to build properly either approach is fine by me.

It would be nice to also have an index somewhere of all the years.

Videos are a bit off-topic and don't need to be dealt with at the same time.

@davidak: I think you have all the data now. Consolidating all the websites is a good idea. If you want to do it then go for it.

**Comment by garbas** [comment]
Here are my proposals with some thoughts:

1. Move nixcon-<year> repositories to NixOS org on GitHub. 

I understand that NixCon is not directly associated with NixOS Foundation but it is part of NixOS community, which is what - I think - the purpose of NixOS org on github us. There will be of course NixCon team with access to all of this repos.

I hope in a long run we will be able to sync all these teams among platforms we use (currently discourse and github).

2. Use Netlify instead of GitHub pages

For all our nixos sites we are currently using Netlify for few reasons:
- allows having preview on pull requests (I can not stress how useful this is)
- if we integrate via github actions you can have any static site generator just triggering nix-build and uploading to Netlify (I think this is a reasonable requirement for any choice a future NixCon organizers)
- manages DNS for us and is just super convinient, 
- does basic analytics OOTB without requiring GDPR popups (it is disabled by default but can be turned on).

In a shell it is really convenient and it does not tie us to the service in any way.

3. Redirects

We must ensure we do the following redirects exists:
- https://nixcon.org -> <CURRENT-YEAR>.nixcon.org (we do this redirect manually once a year)
- https://nixcon.org/<YEAR> -> <YEAR>.nixcon.org
- https://conf.nixos.org -> 2015.nixcon.org
- https://nixcon2017.org -> 2017.nixcon.org
- https://nixcon2018.org -> 2018.nixcon.org

4. Move NixCon video to NixOS youtube channel and create playlist.

A discovery of any sorts of information is far from good in nixos ecosystem. Names Nix and NixOS when searching on google don't help either. It would make a lot of sense to have all this video in one place.

This still means that NixOS Foundation and NixCon foundation are different entities, but we realize we are the same people in both of these organizations and the reasons for split we merely for bureaucratic reasons.

5. archive.nixos.org and S3 bucket where we can keep things archived.

We come across many time when we want to archive something, last example was nix-dev mailing list (which is now under releases.nixos.org/nix-dev/index.html). At this place we could also keep videos. I would select S3 for only reason because we are already using it, I couldn't care really where we store this stuff as long as we don't lose it.


PS. c3voc is a great group! loved their setup.

**Comment by davidak** [comment]
>It would be nice to also have an index somewhere of all the years.

i think https://nixcon.org  is the right place for that with a prominent link to the next conference at the top.

as first step, we can move the list from https://nixos.org/community.html nixcon section to it. it's great to have the video link there because that's what people are probably interested in after the conference is over

>@davidak: I think you have all the data now. Consolidating all the websites is a good idea. If you want to do it then go for it.

I see no benefit when we still have them on different subdomains. It would be identical for visitors but extra work for us. If i understand the discussion correctly, everyone except me wants YEAR.nixcon.org. I really don't like that but can respect different opinions.

>PS. c3voc is a great group! loved their setup.

I also recommend them! I have worked with them many years as camera operator or video mixer at the Chaos Communication Congress and Camp. I actually suggested them in 2015.

**Comment by zimbatm** [comment]
@davidak I think it's a good idea and if you will be implementing this then you also get to decide on prefixes vs sub-domains. What matters is that each year is independent. Ping me on IRC and we can work the finer details together.

**Comment by garbas** [comment]
All the videos from all NixCon's are now in one place https://www.youtube.com/c/NixCon

**Comment by crertel** [comment]
@davidak sorry to be a bother, but is this still something you want to drive forward, or would you mind closing this issue?

**Comment by flyfloh** [comment]
Looks like nobody has the time to work on this and the current situation seems acceptable? I'll close this for now. Feel free to shout at me and we can re-open. :-)
